cpp/DataStorm/sampleFilter/Writer.cpp

```diff
@@ -3,44 +3,85 @@
 #include <DataStorm/DataStorm.h>
 #include <Ice/Ice.h>
 
+#include "TemperatureRange.h"
+
 #include <iostream>
-#include <optional>
 #include <string>
 
 using namespace std;
 
 int
 main(int argc, char* argv[])
 {
-    try
-    {
-        // Instantiates DataStorm node.
-        DataStorm::Node node{argc, argv};
+    // CtrlCHandler must be created before the communicator is created or any other threads are started.
+    Ice::CtrlCHandler ctrlCHandler;
 
-        // Instantiates the "hello" topic. The topic uses strings for keys and values.
-        DataStorm::Topic<string, string> topic{node, "hello"};
+    // Instantiates DataStorm node.
+    DataStorm::Node node{argc, argv};
 
-        // Configure writers to not clear the history. We want the readers to receive all the writer samples.
-        topic.setWriterDefaultConfig({std::nullopt, std::nullopt, DataStorm::ClearHistoryPolicy::Never});
+    // Instantiates the "temperature" topic. The topic uses strings for keys and floats for values.
+    DataStorm::Topic<string, float> topic{node, "temperature"};
 
-        // Instantiate the foo writer and wait for a reader to connect.
-        auto writer = DataStorm::makeSingleKeyWriter(topic, "foo");
-        topic.waitForReaders();
+    // Create a sample filter that only reports temperatures outside of the given temperature range.
+    // The filter criteria TemperatureRange is defined in TemperatureRange.ice Slice file.
```

Comment:
For consistency if you do the other one:
```suggestion
    // The filter criteria `TemperatureRange` is generated from the TemperatureRange.ice Slice file.
```